sqlmesh/core/model/decorator.py

```diff
@@ -119,21 +121,49 @@ def model(
 
         build_env(self.func, env=env, name=entrypoint, path=module_path)
 
+        # Properties to be rendered at model creation time
+        expressions = [exp.Property(this="name", value=exp.to_table(self.name, dialect=dialect))]
+        for field_name in {"enabled", "start", "end"}:
+            if (
+                (field_value := self.kwargs.get(field_name))
+                and isinstance(field_value, exp.Expression)
+                or (isinstance(field_value, str) and field_value.startswith(SQLMESH_MACRO_PREFIX))
+            ):
+                expressions.append(
+                    exp.Property(
+                        this=field_name,
+                        value=parse_one(field_value, dialect=dialect)
+                        if isinstance(field_value, str)
+                        else field_value,
+                    )
+                )
+
+        renderer_kwargs = {
+            "module_path": module_path,
+            "macros": macros,
+            "jinja_macros": jinja_macros,
+            "variables": variables,
+            "path": path,
+            "dialect": dialect,
+            "default_catalog": default_catalog,
+        }
+        meta_renderer = _meta_renderer(
```

Comment:
Perhaps, instead of kwargs, the `_meta_renderer` function should take all the arguments above by name explicitly (no default values).